x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processors/config_driven/types.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ProcessorDefinition } from '@kbn/streams-schema';
+import { configDrivenProcessors } from '.';
+import { WithUIAttributes } from '../../types';
+import type { KvProcessorFormState } from './configs/kv';
+import type { GeoIpProcessorFormState } from './configs/geoip';
+import type { RenameProcessorFormState } from './configs/rename';
+import type { SetProcessorFormState } from './configs/set';
+import type { UrlDecodeProcessorFormState } from './configs/url_decode';
+import type { UserAgentProcessorFormState } from './configs/user_agent';
+
+export type ConfigDrivenProcessorFormState =
+  | KvProcessorFormState
+  | GeoIpProcessorFormState
+  | RenameProcessorFormState
+  | SetProcessorFormState
+  | UrlDecodeProcessorFormState
+  | UserAgentProcessorFormState;
+
+export interface ConfigDrivenProcessorConfiguration<
+  FormStateT,
+  ProcessorDefinitionT extends ProcessorDefinition
+> {
+  value: ConfigDrivenProcessorType;
```

Comment:
I was initially confused by `value` before getting it represents the processor type... should we change it to `type` for simpler understanding?

Reply:
Yeah, can do 👍